server/graphql/v2/mutation/TransactionMutations.ts

```diff
@@ -23,6 +27,92 @@ const transactionMutations = {
       return legacyRefundTransaction(undefined, { id: transaction.id }, req);
     },
   },
+  rejectTransaction: {
+    type: new GraphQLNonNull(Transaction),
+    description: 'Rejects transaction, removes member from Collective, and sends a message to the contributor',
+    args: {
+      transaction: {
+        type: new GraphQLNonNull(TransactionReferenceInput),
+        description: 'Reference of the transaction to refund',
+      },
+      message: {
+        type: GraphQLString,
+        description: 'Message to send to the contributor whose contribution has been rejected',
+      },
+    },
+    async resolve(_, args, req): Promise<typeof Transaction> {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+
+      // get transaction info
+      const transaction = await fetchTransactionWithReference(args.transaction);
+
+      /** refund transaction and set status - - if the transaction has already been
+       * refunded we don't want to try and do it again, but we will continue with
+       * marking the order as 'REJECTED'
+       */
+      let refundedTransaction;
+      if (!transaction.RefundTransactionId) {
+        refundedTransaction = await legacyRefundTransaction(undefined, { id: transaction.id }, req);
+      } else {
+        refundedTransaction = await fetchTransactionWithReference({ legacyId: transaction.RefundTransactionId });
+      }
+      const orderToUpdate = await models.Order.update(
+        {
+          status: orderStatus.REJECTED,
+        },
+        {
+          where: { id: refundedTransaction.OrderId },
+          returning: true,
+        },
+      );
+      if (!orderToUpdate) {
+        throw new NotFound('Order not found');
+      }
+
+      // get membership info
+      const toAccount = await fetchAccountWithReference({ legacyId: transaction.CollectiveId });
+      const fromAccount = await fetchAccountWithReference({ legacyId: transaction.FromCollectiveId });
+      const memberships = await models.Member.findAll({
+        where: {
+          MemberCollectiveId: fromAccount.id,
+          CollectiveId: toAccount.id,
+          role: 'BACKER',
+        },
+      });
+
+      // remove member from Collective
+      if (memberships) {
+        memberships.forEach(membership => {
+          membership.destroy();
+        });
+      }
+
+      // TO DO: email user to let them know their transaction has been rejected
+      const fromAccountEmail = await req.loaders.getUserDetailsByCollectiveId
+        .load(fromAccount.id)
+        .then(user => user.email);
```

Comment:
We don't want to use this strategy for emailing.

Email should go to all collective admins using the standard mechanism we have for that.